packages/config-plugins/src/ios/ProvisioningProfile.ts

```diff
@@ -15,27 +14,27 @@ type ProvisioningProfileSettings = {
   targetName?: string;
   appleTeamId: string;
   profileName: string;
+  buildConfiguration?: string;
 };
 
-function setProvisioningProfileForPbxproj(
+export function setProvisioningProfileForPbxproj(
   projectRoot: string,
-  { targetName, profileName, appleTeamId }: ProvisioningProfileSettings
+  {
+    targetName,
+    profileName,
+    appleTeamId,
+    buildConfiguration = 'Release',
+  }: ProvisioningProfileSettings
 ): void {
   const project = getPbxproj(projectRoot);
-  const target = targetName
+
+  const nativeTargetEntry = targetName
     ? findNativeTargetByName(project, targetName)
     : findFirstNativeTarget(project);
+  const [nativeTargetId, nativeTarget] = nativeTargetEntry;
 
-  if (!target && targetName) {
-    throw new Error(`Unable to find a target named ${targetName}.`);
-  } else if (!target) {
-    throw new Error(`Unable to find any targets in this Xcode project.`);
-  }
-
-  const [nativeTargetId, nativeTarget] = target;
-
-  getBuildConfigurationForId(project, nativeTarget.buildConfigurationList)
-    .filter(([, item]: ConfigurationSectionEntry) => item.buildSettings.PRODUCT_NAME)
+  getBuildConfigurationsForListId(project, nativeTarget.buildConfigurationList)
+    .filter(([, item]: ConfigurationSectionEntry) => item.name === buildConfiguration)
```

Comment:
it's used only on builders so it does not matter much, but you are overwriting signing config for all build configurations